File: skills/openclaw-trustroom/src/tools/list_room_members.ts
// list_room_members: enumerate the current member set of a Trustroom.

import type { JSONRpcClient } from "../client.js";

export interface ListMembersParams {
    room_id: string;
}

export interface ListMembersResult {
    members: { pubkey_hex: string }[];
}

export const listRoomMembers = {
    name: "list_room_members",
    description: "List the current credentialed members of a Trustroom.",
    handler: async (
        params: ListMembersParams,
        ctx: { client: JSONRpcClient },
    ): Promise<ListMembersResult> => {
        // Forward to the Python sidecar via JSONRpcClient.call("list_room_members", params).
        return ctx.client.call<ListMembersResult>("list_room_members", params);
    },
};

File: integration/tools/list_room_members.py
"""list_room_members: enumerate the current member set of a Trustroom."""

from __future__ import annotations

from communication.channel.agent_channel import AgentChannel
from integration.rpc.methods import RpcMethod
from integration.rpc.schema import ListMembersParams, ListMembersResult


class ListRoomMembersMethod(RpcMethod):
    """JSON-RPC method that calls AgentChannel.members."""

    name = "list_room_members"
    params_schema = ListMembersParams
    result_schema = ListMembersResult

    async def __call__(
        self,
        channel: AgentChannel,
        params: ListMembersParams,
    ) -> ListMembersResult:
        # Call channel.members, map AgentId → AgentIdDTO, return list.
        ...
